src/sysusers/sysusers.c

```diff
@@ -486,8 +486,11 @@ static int write_temporary_passwd(const char *passwd_path, FILE **ret_tmpfile, c
                         .pw_gid = i->gid,
                         .pw_gecos = (char*) strempty(i->description),
 
-                        /* "x" means the password is stored in the shadow file */
-                        .pw_passwd = (char*) PASSWORD_SEE_SHADOW,
+                        /* PASSWORD_SEE_SHADOW ("x") means the password is stored in the shadow file.
+                         * We use LOCKED_AND_INVALID ("!*") to let firstboot set the password later.
+                         */
+                        .pw_passwd = (char*) (streq(i->name, "root") ? PASSWORD_LOCKED_AND_INVALID :
+                                                                       PASSWORD_SEE_SHADOW),
```

Comment:
hmm, interesting, so you locked the account in /etc/passwd rather than /etc/shadow.

I think we should not do that, because once people set a password for /etc/passwd with other tools they would do so in /etc/passwd too, and not in /etc/shadow. That's because where the pw is put is orthogonal to setting a password. migration of passwords from one to the other db is done via "pwconv", but other tools typically don't migrate passwords ever...

Hence, I think this line here should stay as it is. But the `struct spwd` initialization should instead get PASSWORD_UNPROVISIONED set for "root", in write_temporary_shadow()